src/utilities/helper.jl

```diff
@@ -73,3 +73,63 @@ function reconstruct!(r, d::MultivariateDistribution, val::AbstractVector, n::In
     r .= val
     return r
 end
+
+struct FlattenIterator{Tname, Tvalue}
+    name::Tname
+    value::Tvalue
+end
+
+Base.length(iter::FlattenIterator) where {T} = _length(iter.value)
```

Comment:
`T` is not defined it seems?
```suggestion
Base.length(iter::FlattenIterator) = _length(iter.value)
```